### PR TITLE
Grafana UI: Improve AutoSaveField component

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**

To add another `InlineToast` to the component to indicate it is saving its input value. Also, place the toast next to the input.

**Why do we need this feature?**

To improve the `AutoSaveField`

**Who is this feature for?**

Everyone using the `AutoSaveField` component

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.tsx [hunk 1]
```diff
@@ -93,22 +85,30 @@ export function AutoSaveField<T = string>(props: Props<T>) {
...
         ref={fieldRef}
+        className={styles.widthFitContent}
       >
         {React.cloneElement(
           children((newValue) => {
             lodashDebounce(newValue);
           })
         )}
       </Field>
+      {isLoading && (
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.tsx:107]: Should this be only on `fieldState.loading`? Is that the one that happens when it is saving?

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Looks cool 🤩

File: packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.tsx (hunk 1 above)
Comment (eledobleefe) [reply packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.tsx:107]: Yes, you're right 😅

**Comment by eledobleefe** [comment]
> When it's on the right side it's right on the border, can we offset it by 2px to the right of the input? if it could form a bit of an arrow on the side it connects to the input would also be cool but not needed :)

Hi @torkelo! Your screenshot is from Storybook, isn't it? Could you tell me how you got there? I cannot see the toast so close, even with the blue border it has more space than yours 🤔 

Captura de pantalla 2023-04-19 a las 17 21 50

**Comment by torkelo** [comment]
@eledobleefe could be chrome 112 on ubuntu issue, in firefox I see proper spacing.

**Comment by eledobleefe** [comment]
> @eledobleefe could be chrome 112 on ubuntu issue, in firefox I see proper spacing.

Hi @torkelo! 
I created an [issue](https://github.com/grafana/grafana/issues/67053) and added it to the @grafana/design-system backlog so we can investigate this behaviour in the future.

**Review by JoaoSilvaGrafana — APPROVED** [review]
Nice work!

**Pull request merged**